apps/whale-api/src/module.api/governance.controller.e2e.ts

```diff
@@ -0,0 +1,328 @@
+import { ListProposalsStatus, ListProposalsType, VoteDecision } from '@defichain/jellyfish-api-core/dist/category/governance'
+import { RegTestFoundationKeys } from '@defichain/jellyfish-network'
+import { Testing } from '@defichain/jellyfish-testing'
+import { MasterNodeRegTestContainer, StartOptions } from '@defichain/testcontainers/dist/index'
+import {
+  GovernanceProposalStatus,
+  GovernanceProposalType,
+  ProposalMasternodeType
+} from '@defichain/whale-api-client/dist/api/governance'
+import { NestFastifyApplication } from '@nestjs/platform-fastify'
+import BigNumber from 'bignumber.js'
+import { createTestingApp, stopTestingApp } from '../e2e.module'
+import { GovernanceController } from './governance.controller'
+
+class MultiOperatorGovernanceMasterNodeRegTestContainer extends MasterNodeRegTestContainer {
+  protected getCmd (opts: StartOptions): string[] {
+    return [
+      ...super.getCmd(opts),
+      `-masternode_operator=${RegTestFoundationKeys[1].operator.address}`,
+      `-masternode_operator=${RegTestFoundationKeys[2].operator.address}`
+    ]
+  }
+}
+const container = new MultiOperatorGovernanceMasterNodeRegTestContainer()
+let app: NestFastifyApplication
+let controller: GovernanceController
+const testing = Testing.create(container)
+let cfpProposalId: string
+let vocProposalId: string
+let payoutAddress: string
+
+describe('governance - listProposals and getProposal', () => {
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+    await testing.container.waitForWalletBalanceGTE(100)
+    await testing.container.call('setgov', [
+      { ATTRIBUTES: { 'v0/params/feature/gov': 'true' } }
+    ])
+    await testing.container.generate(1)
+    app = await createTestingApp(container)
+    controller = app.get(GovernanceController)
+
+    // Create 1 CFP + 1 VOC
+    payoutAddress = await testing.generateAddress()
+    cfpProposalId = await testing.rpc.governance.createGovCfp({
+      title: 'CFP proposal',
+      context: 'github',
+      amount: new BigNumber(1.23),
+      payoutAddress: payoutAddress,
+      cycles: 2
+    })
+    await testing.container.generate(1)
+
+    vocProposalId = await testing.rpc.governance.createGovVoc({
+      title: 'VOC proposal',
+      context: 'github'
+    })
+    await testing.container.generate(1)
+  })
+
+  afterAll(async () => {
+    await stopTestingApp(container, app)
+  })
+
+  // Listing related tests
+  it('should listProposals', async () => {
+    const result = await controller.listProposals()
+    expect(result.data.length).toStrictEqual(2)
```

Comment:
should have at least one test that asserts the proposal shape? e.g. `title`, `context`, etc.
instead of just testing `result.data.length`

Reply:
updated, also added some manipulation because rpc itself doesnt support sorting, data returned is of random order